src/models/IssueTableIndex.gs

```diff
@@ -0,0 +1,394 @@
+    return sheetId + '__' + tableId;
+  },
+
+  /**
+   * @desc Get all stored IssueTable's.
+   * @return {array} Returns array of found IssueTable instances. Empty array if nothing found.
+   */
+  getAll : function () {
+    debug.log('IssueTableIndex_.getAll()');
+
+    var tables = [], sheetId, sheetIdx, tableIndexId, IssueTable;
+
+    // load everything from storage
+    this._load();
+
+    for (sheetId in this._index) {
+      if (this._index.hasOwnProperty(sheetId)) {
+        for (sheetIdx in this._index[sheetId]) {
+          if (this._index[sheetId].hasOwnProperty(sheetIdx)) {
+            tableIndexId = this.tableIndexName(sheetId, this._index[sheetId][sheetIdx]);
+            if (null !== this._tables[tableIndexId]) {
+              IssueTable = new IssueTable_();
+              tables.push(IssueTable.fromJson(this._tables[tableIndexId]));
+            }
+          }
+        }
+      }
+    }
+
+    return tables;
+  },
+
+  /**
+   * @desc Get all stored IssueTable's for passed Sheet Id.
+   * @param {string} sheetId Optional SheetId to fetch tables for. Default current active Sheet.
+   * @return {array} Returns array of found IssueTable objects. Empty array if nothing found.
+   */
+  getAllTablesBySheet : function (sheetId) {
+    sheetId = sheetIdPropertySafe(sheetId);
+    debug.log('IssueTableIndex_.getAllTablesBySheet(%s)', sheetId);
+
+    var tables = [], sheetIdx, tableIndexId, IssueTable;
+
+    // load everything from storage
+    this._load();
+
+    for (sheetIdx in this._index[sheetId]) {
+      if (this._index[sheetId].hasOwnProperty(sheetIdx)) {
+        tableIndexId = this.tableIndexName(sheetId, this._index[sheetId][sheetIdx]);
+        if (null !== this._tables[tableIndexId]) {
+          IssueTable = new IssueTable_();
+          tables.push(IssueTable.fromJson(this._tables[tableIndexId]));
+        }
+      }
+    }
+
+    return tables;
+  },
+
+  /**
+   * @desc Search for a indexed table by its coordinates within a sheet.
+   * @param {string} sheetId Optional SheetId to fetch tables for. Default current active Sheet.
+   * @param {int|Range} columnOrRange A Range instance or column index used to find matching table
+   * @param {int|null} row If column is integer a row index is required as well to identify a cell-coord to search a indexed table for.
+   * @return {FALSE|IssueTable_} If table found it returns a instance of IssueTable_, or FALSE if nothing found.
+   */
+  getTableByCoord : function (sheetId, columnOrRange, row) {
+    debug.log('IssueTableIndex_.getTableByCoord(%s, %s, %s)', sheetId, columnOrRange, row);
+    debug.time('IssueTableIndex.getTableByCoord()');
+
+    var sheetId = sheetIdPropertySafe(sheetId);
+    var column = (typeof columnOrRange === 'object') ? columnOrRange.getColumn() : parseInt(columnOrRange);
+    var row = (typeof columnOrRange === 'object') ? columnOrRange.getRow() : parseInt(row);
+    var sheetIdx, tableIndexId, tableJson, sheet, namedRange;
```

Comment:
Usual javascript style is to initialise variable in the scope they are used.
Nothing wrong with this per se
I'm just more used to seeing
```
for (var sheetIdx in ..) {
}

```
also not sure what the extra "x" is on the end of that variable

Reply:
I "learned" (still JS newbie), that variable declaration should be block vise at beginning of methods. And dynamic declaration inside a function or callable like _for_ may have sideeffect. Dont wanna recall further as im not 100% sure...

The "x" is part of "idx" which is (for me) common abbrivation for "index" = "idx"

Reply:
This is a classic and worth a read - https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882
The book is getting a little old in the tooth but its pretty much the foundation for a lot of modern coding practises.

The principle of declaring the variable as close to the scope as its used increases readability because you only read about the variable when its going to be used. Also it makes it clear to the reader the scope in which it will be used which improves comprehension. (this is not just a javascript thing). Back in the day the compiler / interpreter needed those declarations at the top of the module/method/function so it could reserve space for the variable - this is not needed any more.
Not a release blocker.

Reply:
https://stackoverflow.com/questions/1411463/in-java-should-variables-be-declared-at-the-top-of-a-function-or-as-theyre-ne